Update Loading and Failed Screens for Unsupported Block Editor

SiobhyB opened the pull request:
Fixes the Android side of wordpress-mobile/gutenberg-mobile#3425

`gutenberg`: https://github.com/WordPress/gutenberg/pull/32395
`WordPress-Android`: https://github.com/wordpress-mobile/WordPress-Android/pull/14762
`WordPress-iOS`: https://github.com/wordpress-mobile/WordPress-iOS/pull/16586

## Description

The `/sites/{site_id}/gutenberg` endpoint is [unreliable](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3425#issuecomment-840223555) but was being used to check if Gutenberg was available on a site's backend, and if so, disable the Unsupported Block Editor (UBE).

Since the majority of users will have Gutenberg enabled, it makes sense to remove usage of this unreliable API endpoint. That's done [in the Android PR](https://github.com/wordpress-mobile/WordPress-Android/pull/14762) in this PR chain.

To account for the removal, the loading and failed screens are updated in [the Gutenberg PR](https://github.com/WordPress/gutenberg/pull/32395).

## How has this been tested?

Please refer to [the Gutenberg PR](https://github.com/WordPress/gutenberg/pull/32395) as the central place for the most up-to-date testing instructions.

<hr />

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

SiobhyB committed:
Bump Gutenberg

SiobhyB committed:
Merge branch 'develop' into android/remove-ube-editor-check

SiobhyB committed:
Bump Gutenberg

SiobhyB committed:
Merge branch 'develop' into android/remove-ube-editor-check

SiobhyB committed:
Update RELEASE-NOTES.txt

AliSoftware commented:
Note: The "Build Android RN Bridge & Publish to S3" failure on CI is because pMz3w-diy-p2 got processed by Systems 5h ago which means S3 now returns 404 instead of 403 and we need to [update our `build.gradle` files accordingly](https://github.com/WordPress/gutenberg/blob/3ba2108f636cd68fc16fc4c6564ebed8a487fd30/packages/react-native-aztec/android/build.gradle#L173-L174).
I apparently don't have the permissions to push a patch on the gutenberg repo, but hopefully @oguzkocer does?

SiobhyB committed:
Bump Gutenberg

SiobhyB committed:
Merge branch 'develop' into android/remove-ube-editor-check

SiobhyB committed:
Correct typo in RELEASE-NOTES.txt

This commit corrects the reference to an "in-progress" screen to "failed".

SiobhyB committed:
Bump Gutenberg

SiobhyB committed:
Bump Gutenberg

SiobhyB committed:
Merge branch 'develop' into android/remove-ube-editor-check

SiobhyB committed:
Bump Gutenberg

SiobhyB committed:
Bump Gutenberg

SiobhyB committed:
Merge branch 'develop' into android/remove-ube-editor-check

SiobhyB committed:
Bump Gutenberg

SiobhyB committed:
Merge branch 'develop' into android/remove-ube-editor-check

SiobhyB committed:
Bump Gutenberg

SiobhyB committed:
Merge branch 'develop' into android/remove-ube-editor-check

AmandaRiu reviewed (approved):
Changes LGTM!  Ready to merge once gutenberg is updated. :shipit:

SiobhyB committed:
Bump Gutenberg

SiobhyB committed:
Merge branch 'develop' into android/remove-ube-editor-check

SiobhyB committed:
Update RELEASE-NOTES.txt

This commit moves these changes from the "1.56.0" section of the notes to the "unreleased" section.

SiobhyB committed:
Bump Gutenberg

SiobhyB committed:
Bump Gutenberg

SiobhyB committed:
Bump Gutenberg
